PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -539,18 +517,31 @@ function appendRuleDocUrls(sarifReport: string) {
     return JSON.stringify(sarifJson);
 }
 
-function verifyDTPService() {
-    let url = dtpBaseUrl + "grs/api/v1/dtpServices";
-    return axios.default.get(url, {
-        httpsAgent: httpsAgent,
-        auth: {
-            username: dtpUsername,
-            password: dtpPassword
-        }
-    }).then(() => {
-        isDTPServiceAvailable = true;
-    }).catch((error) => {
-        isDTPServiceAvailable = false;
-        tl.warning("Unable to connect to DTP to retrieve documentation for rules using the provided settings: Error code " + (error.response ? error.response.status : undefined));
-    });
+function verifyDtpRuleDocsService() {
+    return tryToGetNotExistingRuleDocUrl()
+            .then(() => {
+                // Should never reach this block unless there is a `notExistingRule` rule id
+                // and `notExistingAnalyzerId` analyzer id pair in the future.
+                isDtpRuleDocsServiceAvailable = true;
+            }).catch((error) => {
+                const status =  error && error.response && error.response.data ? error.response.data.status : undefined;;
+                if (status == 404) {
+                    isDtpRuleDocsServiceAvailable = true;
+                } else if (status == 401) {
+                    // Need auth to get doc url for DTP version below 2023.1.
+                    isDtpRuleDocsServiceAvailable = false;
+                    tl.warning("Unable to retrieve documentation for rules from current version of DTP. DTP 2023.1 or newer version is needed.");
+                } else {
+                    isDtpRuleDocsServiceAvailable = false;
+                    tl.warning("Unable to connect to DTP to retrieve documentation for rules using the provided settings: Error code " +
+                                status + ". Check the dtp.* values in " + localSettingsPath);
+                }
+            });
+}
+
+/**
+ * 404 is expected when response is returned as normal.
+ */
+function tryToGetNotExistingRuleDocUrl() {
```

Comment:
If this method is only used once, I think it's better to make the call directly in "verifyDtpRuleDocsService" and just add reasonable comment.